### PR TITLE
feat: added spacing tokens in REM

### PR DESCRIPTION
**Pull request by GutierrezGo**
References:
> Jira https://autoricardo.atlassian.net/browse/DM-169

> Figma https://www.figma.com/file/WmNmJ7jdB0Z7tOjj516nFB/AS24-Tokens?node-id=3%3A4

> Figma tokens https://github.com/smg-automotive/figma-tokens/blob/main/tokens.json

## Motivation and context

Setup of base theme SPACING tokens as defined in the design system.

## Take into consideration
* I found a mistake in the figma tokens json. I checked with Sergiu and he already fixed, now the key is `2xl` instead `xxl` and it is in sync with figma as well. 
* I was checking about the unit that we'll use in the spacing tokens case. And after checking with Sergiu we decided to go with `REM` for this case, for the font-size and line-height. If we use `PX` the values don't scale with browser font-size adjustments. Please let me know if I need to bring this topic into a discussion or if we are fine merging this tokens with the `REM` unit.

## Before

No base theme for the spacing tokens

## After

Base theme for spacing tokens 🔭

### CODE REVIEW

File: src/theme/stories/spacingShowcase.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+const SpacingShowCase: FC<Props> = ({ name }) => {
+  const theme = useTheme();
+  const unitRem = parseFloat(theme.spacing[name]);
+  const unitPixel = unitRem / 0.0625;
+  return (
+    <Tr border="1px" borderColor={theme.colors.gray[100]}>
+      <Td>{name}</Td>
+      <Td>{`${unitRem}rem`}</Td>
+      <Td>{`${unitPixel}px`}</Td>
```
Comment (talamcol) [inline src/theme/stories/spacingShowcase.tsx:16]: This is very confusing to me. `rem` has the advantage that it considers the user's font size preference. So this value is correct if you have "medium" but wrong if I increase my font size to large or extra-large. Maybe you can use `getComputedStyle` to get the correct px size?

**Review by talamcol — COMMENTED** [review]
It looks good to me besides the `px` value. Would be cool if you can quickly look into it. Thanks!

File: src/theme/stories/spacingShowcase.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (GutierrezGo) [reply src/theme/stories/spacingShowcase.tsx:16]: I'm not sure if it's possible to displays the `rem` values in figma. In the meantime I'd like to keep the Pixels column and for the confusion, I will apply Colin idea/suggestion about adding a text clarifying that the `px` values can change in the computed tab depending on the browser settings, and the `px` column is ONLY a reference table. What do you think?

[the remark this answers is not in the document]
Comment (GutierrezGo) [reply src/theme/stories/spacingShowcase.tsx:16]: So, I double checked and Figma doesn't display values in `rem`

[the remark this answers is not in the document]
Comment (Averethel) [reply src/theme/stories/spacingShowcase.tsx:16]: Thanks for taking a look, let's roll with what you discussed with Colin, looks good to me 👍🏻

Comment (GutierrezGo) [reply src/theme/stories/spacingShowcase.tsx:16]: As we discussed,  I will add a text where I explain that this is ONLy a reference table, then we won't have more confusions. Thank you for the idea/suggestion

**Pull request merged**